### ISSUE TITLE
Write new VCF files

### ISSUE THREAD

**Issue by mflevine** [brentp/hts-nim#30]
Hi, currently I can not write a brand new VCF. Do you have plans to be able to initialize an empty VCF header and empty Variant objects that can be modified and then written?

**Comment by brentp** [comment]
hi, it's possible to write a VCF. For example in the tests: https://github.com/brentp/hts-nim/blob/master/tests/vcftest.nim#L33

and you can see this in [duphold code](https://github.com/brentp/duphold/blob/master/src/duphold.nim#L554-L591) as well.

There's another example here: https://gist.github.com/brentp/d251b7376961a72143f5efd5b6b66fec

Briefly, you just open with `mode="w"` and then use `copy_header` and `write_header` once, and `write_variant` for each variant.

Please let me know any questions/problems you have.

**Comment by mflevine** [comment]
Hi, thanks for the speedy reply. As far as I can tell, all these examples open an existing VCF copy the header and add it to the new VCF and do the same for the variants. I am having trouble creating a new VCF with new Variants from scratch.

**Comment by brentp** [comment]
ok. here's a full example that does not use another vcf as a template. It should work with recent versions, but I just pushed a small change that makes it give you an better warning if you do it incorrectly:
```Nim
import hts/vcf

var h:Header

h.from_string("""##fileformat=VCFv4.2
##FORMAT=<ID=DP,Number=1,Type=Integer,Description="total reads covering this site">
##INFO=<ID=DP,Number=1,Type=Integer,Description="total reads covering this site">
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	mysample""")

doAssert h.add_string("##FORMAT=<ID=AD,Number=2,Type=Integer,Description=\"ref, alt\"") == Status.OK

var ovcf:VCF

if not open(ovcf, "out.vcf", mode="w"):
  quit "could not open vcf"

echo $h

ovcf.copy_header(h)
ovcf.header = h

var vr : Variant
var vstring = "1	22	rs123	A	T	22.2	PASS	DP=33	GT:DP:AD	0/1:33:15,18"
vr.from_string(ovcf.header, vstring)


doAssert ovcf.write_header
doAssert ovcf.write_variant(vr)

ovcf.close()
```

**Comment by mflevine** [comment]
Hmm this works but it feels a bit wrong to be writing the lines manually.

**Comment by brentp** [comment]
I agree. One could abstract this out and make it nicer, but I haven't had the need just yet. 
htslib, which underlies hts-nim, does not provide much to help here...

**Comment by brentp** [comment]
calling this closed by #32 (and hopefully a soon-to-exist sugar repo from @mflevine )

**Comment by mflevine** [comment]
Added here: https://github.com/mflevine/hts-nim-sugar
